pittapi/dining.py

```diff
@@ -1,98 +1,146 @@
+    if type(date) == datetime.datetime:
+        url = url + date.strftime("%Y-%m-%d")
+    else:
+        url += date
+
+    loc_hours = requests.get(url).json()
+
+    hours = {}
+
+    # location_name : [ [date, status, open, close], [], [], ... ]
+    for i in loc_hours["the_locations"]:
+        week_info = []  # week info contains list of [date, status, open time, close time]
+        for j in i["week"]:
+            date = j["date"]
+
+            if j["status"] == "closed":  # to handle so locations having hours despite being closed
+                start_time = 0
+                end_time = 0
             else:
-                data["name"] = fields["title"]
-            data["status"] = fields["status"]
-            try:
-                data["hours"] = fields["eventDate"]["formatted"]
-            except TypeError:
-                data["hours"] = "unavailable"
-
-            dining_locations.append(data)
-
-    return dining_locations
-
-
-# def get_location_by_name(location):
-#    try:
-#        return get_locations()[location]
-#    except:
-#        raise ValueError('The dining location is invalid')
-
-# def get_location_menu(location=None, date=None):
-# location can only be market, market's subordinates, and cathedral cafe
-# if location is none, return all menus, and date will be ignored
-# date has to be a day of the week, or if empty will return menus for all days of the week
-#
-# https://www.pc.pitt.edu/dining/menus/flyingStar.php
-# https://www.pc.pitt.edu/dining/menus/bellaTrattoria.php
-# https://www.pc.pitt.edu/dining/menus/basicKneads.php
-# https://www.pc.pitt.edu/dining/menus/basicKneads.php
-# https://www.pc.pitt.edu/dining/menus/magellans.php
-# https://www.pc.pitt.edu/dining/locations/cathedralCafe.php
-#    return []
+                start_time = datetime.time(
+                    j["hours"][0]["start_hour"], j["hours"][0]["start_minutes"])
+                end_time = datetime.time(
+                    j["hours"][0]["end_hour"], j["hours"][0]["end_minutes"])
+
+            week_info.append([datetime.datetime(int(date[0:4]), int(
+                date[5:7]), int(date[8:])), j["status"], start_time, end_time])
+
+        hours[i["name"]] = week_info
+
+    return hours
+
+
+def get_menu(location, date=datetime.datetime.today()):
+    """
+    location will be string of location name
+    date must be in YYYY-MM-DD format
+
+    Returns:
+        None if location is closed
+        
+
+    Raises Exception:
+        If req status is unsuccessful
+        Usually occurs when no menu is found
+    """
+    menu = {}
+    url = "https://api.dineoncampus.com/v1/location/"
+
+    loc_id = get_all_locations()[location]
+
+    url += loc_id
+    url += "/periods?platform=0&date="
+
+    # could probably restrict arguments
+    if(type(date) == datetime.datetime):
+        date = date.isoformat()
+        url += date
+    else:
+        url += date
```

Comment:
```suggestion
    if isinstance(date, datetime.datetime):
        date = date.isoformat()
    url += date
```

You have `url += date` in both branches of the if-else block so that line of code will always run